src/carp/helpers/Map.ts

```diff
@@ -0,0 +1,54 @@
+import MapObject from '../types/MapObject';
+
+export default class Map implements MapObject {
+  public width: number;
+
+  public height: number;
+
+  public enter: { x: number; y: number; room_id: number };
+
+  public exit: { x: number; y: number; room_id: number };
+
+  public room_count: number;
+
+  public rooms: {
+    [id: string]: {
+      doors: [];
+      neighbors: [];
+      id: number;
+      top: number;
+      left: number;
+      width: number;
+      height: number;
+      deadend: boolean;
+    };
+  };
+
+  public world: [[]];
+
+  public toJSON(): string {
+    const { width, height } = this;
+    return JSON.stringify({ width, height });
```

Comment:
Why are you only storing the width/height when converting to JSON? Why not the same fields as fromJSON?

Reply:
Good catch, fixed it.